### ISSUE TITLE
Exit code !== 0 doesn't necessarily mean the promise should be rejected

### ISSUE THREAD

**Issue by gunderson** [patrick-steele-idem/child-process-promise#12]
Specifically in my case, ImageMagick compare exits with 0 when images are similar, exits with 1 when they are dissimilar and with 2 on error.

http://stackoverflow.com/questions/30224018/imagemagick-compare-exits-with-code-1-when-spawned-from-node-but-exits-with-co#30224018

**Comment by patrick-steele-idem** [comment]
In general, a non-zero exit code means that something went wrong in the spawned process. However, any meaning can be assigned to an exit code. I recommend the following:

```javascript
var spawn = require("child-process-promise").spawn,
    filenameA = "img0.png",
    filenameB = "img1.png",
    filenameO = "img.0-1.png";

var p = spawn('compare', [filenameA, filenameB, '-fuzz', 20, '-highlight-color', "#ffffff", '-lowlight-color', "#000000", filenameO])
    .then(function(){
        return {
            similar: true
        };
    })
    .fail(function (err) {
        return {
            similar: false
        };
    });
```

Does that work for you?

**Comment by gunderson** [comment]
Thank you for taking the time to respond. In this specific case it does hack the failure to work as the imagemagick devs expect, but in practice, it will cause complex chains to fail. I think a better solution would be to have the exit-0 assumption be configurable. I'll make a PR.

**Comment by patrick-steele-idem** [comment]
New version published: `child-process-promise@1.1.0`
